### PR TITLE
Update Memcached Metric Names

### PR DESCRIPTION
**Pull request by Mrod1598**
**Description:** <Describe what has changed.>
Update Memcached metric names to better follow convention.

### CODE REVIEW

**Review by djaglowski — APPROVED** [review]
I agree with the changes here. Generally we should try to avoid breaking changes, but the component is still beta and there are not semantic conventions for these metrics yet, so I support this as a step in the right direction.

File: receiver/memcachedreceiver/documentation.md [hunk 1]
```diff
@@ -10,15 +10,15 @@ These are the metrics available for this scraper.
...
-| memcached.current_connections | The current number of open connections. | connections | Sum | <ul> </ul> |
+| memcached.connections.current | The current number of open connections. | connections | Sum | <ul> </ul> |
+| memcached.connections.total | Total number of connections opened since the server started running. | connections | Sum | <ul> </ul> |
 | memcached.current_items | Number of items currently stored in the cache. | 1 | Sum | <ul> </ul> |
 | memcached.evictions | Cache item evictions. | 1 | Sum | <ul> </ul> |
 | memcached.network | Bytes transferred over the network. | by | Sum | <ul> <li>direction</li> </ul> |
 | memcached.operation_hit_ratio | Hit ratio for operations, expressed as a percentage value between 0.0 and 100.0. | % | Gauge | <ul> <li>operation</li> </ul> |
 | memcached.operations | Operation counts. | 1 | Sum | <ul> <li>type</li> <li>operation</li> </ul> |
-| memcached.rusage | Accumulated user and system time. | 1 | Sum | <ul> <li>state</li> </ul> |
+| memcached.rusage.cpu | Accumulated user and system time. | 1 | Sum | <ul> <li>state</li> </ul> |
```
Comment (bogdandrutu) [inline receiver/memcachedreceiver/documentation.md:20]: @djaglowski should this be "memcached.cpu.usage" ?

Comment (djaglowski) [reply receiver/memcachedreceiver/documentation.md:20]: Yes, you're right. 

@Mrod1598 ^

**Review by bogdandrutu — COMMENTED** [review]
Please add a changelog entry

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**